lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/GroupCommClient.cs

```diff
@@ -84,11 +85,15 @@ public void WaitingForRegistration(CancellationTokenSource cancellationSource =
                     group.WaitingForRegistration(cancellationSource);
                 }
             }
-            catch (SystemException e)
+            catch (OperationCanceledException)
```

Comment:
That's right. Since we re-throw, we should catch and dispose for all the exceptions.